apps/banners/middleware.py

```diff
@@ -0,0 +1,23 @@
+import re
+
+
+class BannerLocationMiddleware:
```

Comment:
you could instead use `request.resolver_match.view_name` which is the name of the view including the namespace e.g. `experiments:experiments_home`


https://docs.djangoproject.com/en/5.2/ref/urlresolvers/#django.urls.ResolverMatch.view_name

Reply:
If you're using the view name is it really necessary to have this middleware? You could just use the view name in directly without mapping it to 'banner_location'